core/os/os.cpp

```diff
@@ -285,6 +285,20 @@ String OS::get_resource_dir() const {
 	return ProjectSettings::get_singleton()->get_resource_path();
 }
 
+// Absolute path to tmp://
+#if !defined(__clang__) && (__cplusplus >= 201703L || (defined(_MSVC_LANG) && _MSVC_LANG >= 201703L))
+String OS::get_temporary_dir() const {
+	// C++17 or greater is required for standard temporary directories.
```

Comment:
There's no need for this check, we use C++17 by default.

Reply:
Some ancient versions? Xcode on 10.14+ should support `std::filesystem`. In general, we do not use `std` stuff unless absolutely necessary, so it might be better to use env vars on Unix and `GetTempPathW` on Windows.
```c++
	const char *env_paths[] = {"TMPDIR", "TMP", "TEMP", "TEMPDIR"};
	const char *ret = nullptr;
	for (const char *ep : env_paths) {
		if ((ret = getenv(ep))) {
			break;
		}
	}
  	if (ret == nullptr) {
    		ret = "/tmp";
	}
```
This is pretty much exactly what LLVM default `std::filesystem::temp_directory_path()` do.

Reply:
Yeah its version 10.15. I left a comment about it:
//... -mmacosx-version-min=10.13 to -mmacosx-version-min=10.15 would allow building this part with clang after clang version 5.
However the build tests for Godot use 10.13 for some reason.

Reply:
To be able to support very old versions and avoiding the std library may be a good enough reason to write it that way. I'll give it a shot. THanks for the feedback.

Reply:
It is done! Usage of std library filesystem call(s) have been replaced by platform specific functions. Hopefully this will remove the requirement for C++17 in the first place so no more #if defs and no more Clang errors... Hopefully.